### PR TITLE
Feature/game logic

### PR DESCRIPTION
**Pull request by Ryguy-1**
Added the ResourceLogic.cs script which manages resource logic so Money, Paper Supply, and Citizen Happiness.

### CODE REVIEW

File: Assets/Scripts/ResourceLogic.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        paperSupply = 50;
+        citizenHappiness = 50;
+        money = 50;
+        lengthOfGame = 2;
+        hasSetLength = false;
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (ChaoticLeah) [inline Assets/Scripts/ResourceLogic.cs]: Update could be removed

**Review by ChaoticLeah — CHANGES_REQUESTED** [review]
The script seems ok, you could remove the update() method though.

File: Assets/Scripts/MenuNavigation.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ public void StartGame()
     {
         SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex+1);
         Debug.Log("Game Started");
+
```
Comment (kronodeus) [inline Assets/Scripts/MenuNavigation.cs]: It is better to avoid making unnecessary changes to files, since they could result in merge conflicts. It is probably better to undo this change to prevent that

File: Assets/Scripts/ResourceLogic.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+
+    // Start is called before the first frame update
+    void Awake()
+    {
+        paperSupply = 50;
+        citizenHappiness = 50;
+        money = 50;
+        lengthOfGame = 2;
+        hasSetLength = false;
```
Comment (kronodeus) [inline Assets/Scripts/ResourceLogic.cs]: Is there a reason we are initializing these here instead of where they are declared? (above)

File: Assets/Scripts/MenuNavigation.cs (hunk 2 above)
Comment (Ryguy-1) [reply Assets/Scripts/MenuNavigation.cs]: Ok. How should I go about removing changes to these files?

File: Assets/Scripts/ResourceLogic.cs (hunk 3 above)
Comment (Ryguy-1) [reply Assets/Scripts/ResourceLogic.cs]: I was treating it like a constructor. Not exactly sure if that is what it should be used for though, so this could not be best practice in unity. Would you recommend I change it?

File: Assets/Scripts/ResourceLogic.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+public class ResourceLogic : MonoBehaviour
+{
+    //values are out of 100
+    private int paperSupply;
+    private int citizenHappiness;
+    private int money;
+    //value is out of 5 with a HIGHER VALUE = SHORTER GAME!!
+    private int lengthOfGame;
+    private Boolean hasSetLength;
```
Comment (eanewman) [inline Assets/Scripts/ResourceLogic.cs]: Fun fact: the primitive version of Boolean in C# is bool (you could make this a bool since you don't need the object version)

File: Assets/Scripts/ResourceLogic.cs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+    }
+
+
+
+    //MODIFY VALUES SHOULD BE -3  ->  3 TO MAKE GAME LENGTH APPROPRIATE
+
+    public void modifyPaperSupply(int amount)
+    {
+        if (amount<0 && paperSupply + amount * lengthOfGame >= 0)
```
Comment (eanewman) [inline Assets/Scripts/ResourceLogic.cs]: This calculation is done many times in the code below - sometimes it helps teammates (and your future self) follow your logic when you extract this to a variable and name it descriptively (so people aren't concentrating as much on how the calculation is determined vs what it represents).

Comment (Ryguy-1) [reply Assets/Scripts/ResourceLogic.cs]: Sorry, I am a little confused what you mean by this. Is there any way you could clarify what I should extract to a variable and how you would recommend I do that most effectively? Thank you!

File: Assets/Scripts/ResourceLogic.cs (hunk 4 above)
Comment (Ryguy-1) [reply Assets/Scripts/ResourceLogic.cs]: that's a great idea!

File: Assets/Scripts/ResourceLogic.cs (hunk 5 above)
Comment (ChaoticLeah) [reply Assets/Scripts/ResourceLogic.cs]: Probably make a function for it I'm guessing?

File: Assets/Scripts/MenuNavigation.cs (hunk 2 above)
Comment (eanewman) [reply Assets/Scripts/MenuNavigation.cs]: You have a multiple options. Here are some I can think of now:

1. Since this has the line number, one way is to just remove the exact line that was added in your editor locally (commit that locally and push to your remote branch and It'll show up here in this pull request)
2. Another (more involved) way to do this is to revert all changes in this file. Figure out which commit changed this file, grab the hash of the commit just before it, and use it with the following command: `git checkout <enter-commit-hash-here> -- file1/to/restore` 
3. You could probably just remove the whitespace directly via GitHub, but I don't recommend that in general because it is very prone to error (you'd never want to do that with code).

I'd use the first method for this case. The second is just in case you're curious about how to handle more complicated situations.

**Review by ChaoticLeah — APPROVED** [review]
Looks Good!
